### PR TITLE
Quiet some warnings

### PR DESCRIPTION
**Pull request by sethalves**
- fix some build warnings
- fix some asan division by zero and uninitialized read warnings
- remove a couple of spammy log lines

### CODE REVIEW

File: libraries/shared/src/MaterialMappingMode.h [hunk 1]
```diff
@@ -13,7 +13,8 @@
 
 enum MaterialMappingMode {
     UV = 0,
-    PROJECTED
+    PROJECTED,
+    UNSET_MATERIAL_MAPPING_MODE
```
Comment (SamGondelman) [inline libraries/shared/src/MaterialMappingMode.h]: This and UNSET_GIZMO_TYPE should probably be at the beginning, I think

Comment (sethalves) [reply libraries/shared/src/MaterialMappingMode.h]: I was afraid that UV being zero might matter someplace.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19764/HighFidelity-Beta-PR15674-07c2d80.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19764/HighFidelity-Beta-Interface-PR15674-07c2d80.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19764/HighFidelity-Beta-PR15674-07c2d80.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19764/HighFidelity-Beta-Interface-PR15674-07c2d80.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19764/HighFidelity-Beta-PR15674-07c2d80.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19764/HighFidelity-Quest-Beta-PR15674-07c2d80.apk)

File: libraries/shared/src/MaterialMappingMode.h (hunk 1 above)
Comment (SamGondelman) [reply libraries/shared/src/MaterialMappingMode.h]: Ah hmmm maybe...if we kept it as is we’d just have to remember to bump the protocol version if we ever added a new one

Comment (sethalves) [reply libraries/shared/src/MaterialMappingMode.h]: I added some comments so that the intent is more clear.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19771/HighFidelity-Beta-PR15674-02ff4ea.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19771/HighFidelity-Quest-Beta-PR15674-02ff4ea.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19771/HighFidelity-Beta-PR15674-02ff4ea.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19771/HighFidelity-Beta-Interface-PR15674-02ff4ea.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19771/HighFidelity-Beta-PR15674-02ff4ea.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19771/HighFidelity-Beta-Interface-PR15674-02ff4ea.exe

**Pull request merged**